libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -229,12 +291,63 @@ internal object MapboxNavigationTelemetry {
         }
         metricsReporter = reporter
         feedbackEventCacheMap.clear()
-
-        registerListeners(mapboxNavigation)
         postTurnstileEvent()
+        telemetryStart()
+        registerListeners(mapboxNavigation)
         log("Valid initialization")
     }
 
+    fun destroy(mapboxNavigation: MapboxNavigation) {
+        telemetryStop()
+        MapboxMetricsReporter.disable()
+        mapboxNavigation.run {
+            unregisterLocationObserver(locationsCollector)
+            unregisterRouteProgressObserver(routeProgressObserver)
+            unregisterRoutesObserver(routesObserver)
+            unregisterOffRouteObserver(offRouteObserver)
+            unregisterNavigationSessionObserver(navigationSessionStateObserver)
+            unregisterArrivalObserver(arrivalObserver)
+        }
+    }
+
+    private fun telemetryStart() {
+        log("sessionStart")
+        telemetryState = when (sessionState) {
+            IDLE -> {
```

Comment:
This will always fall into the `IDLE` branch, won't it? Can we simplify this setup then?

Reply:
I'd say it should be but cannot guaranty. 
NavTelemerty lifecycle(init/destroy) is controlled outside, this way I'd keep handling all possible options and keep it as is